### PR TITLE
Data sources UI improvements

### PR DESCRIPTION
**Pull request by derek-ho**
### Description
Make changes according to UX feedback:
- Change spacing/heading style of permissions sub group on configure flow
- Make CTA for error toast
- Remove edit capabilities on datasource specific page
- hook up s3 and prometheus correctly, and send actual data in request

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/datasources/components/manage/access_control_tab.tsx [hunk 1]
```diff
@@ -64,31 +41,6 @@ export const AccessControlTab = (props: AccessControlTabProps) => {
...
-    http!.put(`${DATACONNECTIONS_BASE}`, {
-      body: JSON.stringify({
-        name: props.dataConnection,
-        allowedRoles: selectedQueryPermissionRoles.map((role) => role.label),
-        connector: props.connector,
-        properties: props.properties,
-      }),
-    });
-    setMode('view');
-  };
```
Comment (derek-ho) [inline public/components/datasources/components/manage/access_control_tab.tsx:90]: datasources is read only for now until backend support PATCH

File: public/components/datasources/components/new/auth_details.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  currentPassword: string;
+  currentUsername: string;
+  currentAccessKey?: string;
+  currentSecretKey?: string;
+  currentRegion?: string;
+  setRegionForRequest?: React.Dispatch<React.SetStateAction<string>>;
+  setAccessKeyForRequest?: React.Dispatch<React.SetStateAction<string>>;
+  setSecretKeyForRequest?: React.Dispatch<React.SetStateAction<string>>;
+  setPasswordForRequest: React.Dispatch<React.SetStateAction<string>>;
+  setUsernameForRequest: React.Dispatch<React.SetStateAction<string>>;
```
Comment (Swiddis) [inline public/components/datasources/components/new/auth_details.tsx:22]: I know we do it quite often so I'm more just curious: is there any way to send updaters in props without having to write out all these dispatch types and keep track of everything?

File: public/components/datasources/components/new/auth_details.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    currentAccessKey,
+    currentSecretKey,
+    currentRegion,
+    currentAuthMethod,
+    setAccessKeyForRequest,
+    setPasswordForRequest,
+    setRegionForRequest,
+    setSecretKeyForRequest,
+    setUsernameForRequest,
+  } = props;
```
Comment (Swiddis) [inline public/components/datasources/components/new/auth_details.tsx:38]: At this point I would probably just use the object directly to make it more explicit, call it `authProps` and use e.g. `authProps.currentUsername` or `authProps.setAccessKeyForRequest`. The extra typing is worth the clarity.

File: public/components/datasources/components/new/review_prometheus_datasource_configuration.tsx [hunk 4]
```diff
@@ -84,7 +86,9 @@ export const ReviewPrometheusDatasource = (props: ConfigurePrometheusDatasourceP
               <EuiFlexItem grow={false}>
                 <EuiText className="overview-title">Authentication method</EuiText>
                 <EuiText size="s" className="overview-content">
-                  {currentUsername ? 'Basic auth' : 'AWSSigV4'}
+                  {currentAuthMethod === 'basicauth'
+                    ? 'Basic authentication'
+                    : 'AWS Signature Version 4'}
```
Comment (Swiddis) [inline public/components/datasources/components/new/review_prometheus_datasource_configuration.tsx:91]: No handling for `noauth`. Seems pervasive in some other spots as well.

**Review by Swiddis — APPROVED** [review]

File: public/components/datasources/components/new/review_prometheus_datasource_configuration.tsx (hunk 4 above)
Comment (derek-ho) [reply public/components/datasources/components/new/review_prometheus_datasource_configuration.tsx:91]: Prometheus only supports basic and sigv4

**Review by ps48 — APPROVED** [review]

**Review by YANG-DB — APPROVED** [review]

**Pull request merged**